### DISCUSSION TITLE
Blocks Running on Sub Folders? Or custom segment implementations?

### DISCUSSION THREAD

**Discussion by terryaney** [JanDeDobbeleer/oh-my-posh#2144] [Troubleshoot] [answered]
Hi, if I have a folder structure like:

```
    +-- Project
    |   +-- .git
    |   +-- Bar.cs
    |   +-- SubProject
    |    |   +-- .git
    |    |   +-- Foo.cs

```

When I'm in the `\Project` folder, the `git` segment works nicely.  What I would like to do is add an additional "block" that displays above Project's prompt, to display a `git` segment for `\SubProject`.  Is that possible?

In reality, I want to create a block for each sub folder that starts with "Shared" and contains a git repositiory.  I have 2-3 at the most, usually only 1 (if one exists).

If not possible, is it possible to create segments that 'plug into' Oh My Posh, so I could write my own custom segment implementation and use that?

**Update**
This is sample prompt I used to use via `posh-git` where I built the prompt myself via the `set-content` method.  I looped directories when inside a git repo via the `Get-ChildItem -Directory -Filter Shared* | ForEach-Object` mechanism.

So `\Evolution` has no Shared* folders while `\NexGen` and `\AnthemSeverance` have 1 and 2 sub repositories respectively.

![image](https://user-images.githubusercontent.com/6237247/164990782-e0802ae6-1223-4109-bfb9-080291e59c9e.png)

**Comment by JanDeDobbeleer** [comment, marked as answer]
Is SubProject a git submodule in this case? That would make it easily discoverable. If not, it's tricky.

**Reply by terryaney** [reply]
Not a 'legit' submodule, but behaving that way more or less I think.  It is just a 'regular' repo, and the parent repo's .gitignore ignores all Shared* folders, so it isn't processed.  So guessing it is tricky...?

**Reply by terryaney** [reply]
What about using the `Command` segment?  Any way to have that somehow call a C# script/assembly and return a string formatted so that Oh My Posh renders it nicely (so that it looks like it is simply a `git` segment?  My shell/scripting skills are non-existent, and I'm not sure if you can change the `shell` property to something that can call something written in C# or not.

**Reply by JanDeDobbeleer** [reply]
It's not the best way. When extending segments, we're always looking for features others can benefit from as well. I wouldn't mind having something in place for submodules, that makes sense, but I personally feel this setup is a bit of anti pattern (thus not very common).

I'll think about the best way to resolve this. But a custom executable might just do the trick (but we don't have that segment just yet).

**Reply by terryaney** [reply]
Quick question...not sure if you will know or not.  But if I'm just trying to do this extra feature myself by using `Write-Host` in a PowerShell function...do you know how to make the foreground color transparent so I could reproduce OMP behavior of the powerline with an empty 'previous segment'?

**Reply by JanDeDobbeleer** [reply]
That requires writing ANSI as that's not possible by using `Write-Host` and the standard API. 

PowerShell 7 and later:
```
Write-Output "`e[97;7m Transparent on white `e[m"
```
Windows PowerShell:
```
Write-Output "$([char]27)[97;7m Transparent on white $([char]27)[m"
```

**Comment by terryaney** [comment]
Thanks.  I saw that example somewhere last night on the web.  How would I do transparent on ‘yellow’?  That is what I couldn’t figure out how to do, specify a custom background color.

Sent from Mail for Windows

**Reply by JanDeDobbeleer** [reply]
I suggest you read [this](https://www.lihaoyi.com/post/BuildyourownCommandLinewithANSIescapecodes.html) post :-)
